implementations/rust/ockam/ockam_identity/src/channel/decryptor_worker.rs

```diff
@@ -29,18 +32,7 @@ use ockam_node::{Context, WorkerBuilder};
 use tracing::{debug, info, warn};
 
 pub(crate) struct DecryptorWorker {
-    role: Role,
-    addresses: Addresses,
-    // Route to the other side of the channel
-    remote_route: Route,
-    remote_backwards_compatibility_address: Option<Address>,
-    init_payload: Option<Vec<u8>>,
-    identity: Identity,
-    trust_policy: Arc<dyn TrustPolicy>,
-    state_key_exchange: Option<KeyExchange>,
-    state_exchange_identity: Option<ExchangeIdentity>,
-    state_initialized: Option<Initialized>,
-    plaintext_session_id: Option<SessionId>,
+    state: Option<State>,
```

Comment:
Isn't the `state` always `Some`? A `Decryptor` should always have some state, isn't it?

Reply:
This is actually needed in order to allow a "switch" from one state to the other: during the passage we need to move some instances and a "invalid state" need to exist, either inside the state or as a None

Reply:
I did implement it with an `Invalid` state within State in the beginning, I simply thought I was more elegant using an Option.
You need to use `alloc::mem::replace(state,Invalid)`, create the new state by moving instances, and then update the state.

Reply:
I've written a small sample of what I did with the `Invalid`, as you can see, it's far from being elegant :(

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f0896128920c1aa65f296be85ddd8459